### PR TITLE
Enable `capture_func_transforms` by default

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123357
* #123268
* #122213
* #122212
* __->__ #122211

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
sure, but what's up with the test failures?

**Comment by guilhermeleobas** [comment]
Just as a note to myself. The following tests are now failing with PYTORCH_TEST_WITH_DYNAMO enabled.

```
ERROR: test_vmap_chunksize_composition_in_dim_0_out_dim_1_randomness_error (__main__.TestVmapOperators)
ERROR: test_vmap_chunksize_composition_in_dim_1_out_dim_0_randomness_error (__main__.TestVmapOperators)
ERROR: test_vmap_chunksize_composition_in_dim_1_out_dim_1_randomness_error (__main__.TestVmapOperators)
ERROR: test_vmap_chunksize_composition_in_dim_0_out_dim_0_randomness_error (__main__.TestVmapOperators)
ERROR: test_cat (__main__.TestVmapOperators)
ERROR: test_one_hot (__main__.TestVmapOperators)
ERROR: test_new_empty_strided (__main__.TestVmapOperators)
ERROR: test_chunk (__main__.TestVmapOperators)
ERROR: test_view_as_complex (__main__.TestVmapOperators)
ERROR: test_unbind (__main__.TestVmapOperators)
ERROR: test_stack (__main__.TestVmapOperators)
ERROR: test_roll_no_dims (__main__.TestVmapOperators)
```

File: test/functorch/test_eager_transforms.py [hunk 1]
```diff
@@ -4786,7 +4786,7 @@ def predict(params_and_buffers, x):
     # torch.compile is not supported on Windows or on Python 3.12+
     @expectedFailureIf(IS_WINDOWS or (sys.version_info >= (3, 12)))
     @torch._dynamo.config.patch(suppress_errors=False)
-    @torch._dynamo.config.patch(capture_func_transforms=True)
+    @skipIfTorchDynamo("Do not test torch.compile on top of torch.compile")
```
Comment (guilhermeleobas) [inline test/functorch/test_eager_transforms.py:4789]: remove this one?

**Comment by guilhermeleobas** [comment]
There's now quite a bit of failures on `test_eager_transforms.py` when running with dynamo:

```
* ERROR: test_vmap_functionalize_jvp_cpu (__main__.TestFunctionalizeCPU)
NYI: querying is_contiguous inside of vmap for memory_format other than torch.contiguous_format


* ERROR: test_hessian_vectorize_correctness_simple_cpu (__main__.TestHessianCPU)
* ERROR: test_against_reference_simple_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_argnums_defaults_to_zero_jacfwd_cpu (__main__.TestJacCPU)
* FAIL: test_complex_error_cpu (__main__.TestJacCPU)
* ERROR: test_multiple_outputs_single_argnums_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_multiple_outputs_pytree_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_multiple_outputs_multiple_argnums_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_multiple_inputs_pytree_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_multiple_inputs_outputs_pytree_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_argnums_defaults_to_zero_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_against_reference_simple_jacfwd_cpu (__main__.TestJacCPU)
* ERROR: test_hessian_vectorize_correctness_simple_cpu (__main__.TestHessianCPU)

aten::alias() Expected a value of type 'Tensor' for argument 'self' but instead found type 'int'.



* ERROR: test_jacfwd_different_levels_cpu (__main__.TestHessianCPU)
InferenceMode::is_enabled() && self.is_inference() INTERNAL ASSERT FAILED at "/home/guilhermeleobas/git/pytorch/aten/src/ATen/native/VariableMethodStubs.cpp":66, please report a bug to PyTorch. Expected this method to only be reached in inference mode and when all the inputs are inference tensors. You should NOT call this method directly as native::_fw_primal. Please use the dispatcher, i.e., at::_fw_primal. Please file an issue if you come across this error otherwise.


* FAIL: test_maml_regression_mechanism_make_functional_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_maml_regression_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_FeatureAlphaDropout_mechanism_make_functional_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_FeatureAlphaDropout_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_Dropout_mechanism_make_functional_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_Dropout_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_Dropout_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_find_learning_rate_ensembling_AlphaDropout_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_ensemble_regression_mechanism_make_functional_cpu (__main__.TestExamplesCorrectnessCPU)
* FAIL: test_ensemble_regression_mechanism_functional_call_cpu (__main__.TestExamplesCorrectnessCPU)
assert name in self.cell_and_freevars()


* FAIL: test_requires_grad_inside_transform_cpu (__main__.TestComposabilityCPU)
* FAIL: test_autograd_function_no_setup_context_transform_jacfwd_cpu (__main__.TestComposabilityCPU)
* FAIL: test_autograd_function_no_setup_context_transform_hessian_cpu (__main__.TestComposabilityCPU)
AssertionError: RuntimeError not raised


* ERROR: test_linearize_composition_cpu_float32 (__main__.TestLinearizeCPU)
'TreeSpec' object has no attribute 'type'
Fixed in https://github.com/pytorch/pytorch/pull/123118


* ERROR: test_take_jacrev_cpu (__main__.TestJacCPU)
* ERROR: test_take_jacfwd_cpu (__main__.TestJacCPU)
vmap: aten::put_(self, *extra_args) is not possible because there exists a Tensor `other` in extra_args that has more elements than `self`. This happened due to `other` being vmapped over but `self` not being vmapped over at level 1. Please try to use out-of-place operators instead of aten::put_. If said operator is being called inside the PyTorch framework, please file a bug report instead.
```

**Review by zou3519 — APPROVED** [review]
LGTM

**Pull request closed without merging**